rust/noosphere/src/error.rs

```diff
@@ -11,9 +14,19 @@ pub enum NoosphereError {
 
     #[error("Missing configuration: {0}")]
     MissingConfiguration(&'static str),
+}
 
-    #[error("{0}")]
-    Other(anyhow::Error),
+impl NoosphereError {
+    /// An error code representing the error. These mappings should never
+    /// change between versions and should be append only!
```

Comment:
I don't think this is the correct contract to set with the caller. All API is subject to change between versions, including which numbers map to which error codes. If the change is a breaking one (e.g., we change an error code), the next release will have the appropriate Semver increment.

It would be capricious to break the API contract without good reason, so the chance of these things changing is relatively low. But, we definitely reserve the right to change these codes and their meaning over time, as called for by future research, development and discovery.

Reply:
One possible remedy to instability in these codes: export their values as constants. That way, a caller can compare against a named constant that can update over time as these codes update.

Reply:
Relaxing the ability to not change sounds good, and same with exposing constants :+1: